src/main/scala/com/microsoft/hyperspace/actions/OptimizeAction.scala

```diff
@@ -141,15 +141,22 @@ class OptimizeAction(
     val hadoopConf = spark.sessionState.newHadoopConf()
     val absolutePath = PathUtils.makeAbsolute(indexDataPath.toString, hadoopConf)
     val newContent = Content.fromDirectory(absolutePath, fileIdTracker, hadoopConf)
-    val updatedDerivedDataset = previousIndexLogEntry.derivedDataset.copy(
-      properties = previousIndexLogEntry.derivedDataset.properties
-        .copy(
-          properties = Hyperspace
-            .getContext(spark)
-            .sourceProviderManager
-            .getRelationMetadata(previousIndexLogEntry.relations.head)
-            .enrichIndexProperties(
-              prevIndexProperties + (IndexConstants.INDEX_LOG_VERSION -> endId.toString))))
+    val updatedDerivedDataset = {
+      previousIndexLogEntry.derivedDataset match {
+        case hashPartitionIndex: HyperSpaceIndex.HashPartitionIndex =>
```

Comment:
What other cases will there be available? Can you put them in the PR description?